### PR TITLE
Jvenstad/tenant cd

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth please review.

### CODE REVIEW

**Comment by jonmv** [comment]
Problem: a version is pushed to a system and applications upgrade to it, but the version isn't released to, e.g., Maven Central. Controller later gives that version a compile version for application to build against. Doesn't work. This fixes it. Also filters out `broken` confidence versions.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/VersionStatusSerializer.java [hunk 1]
```diff
@@ -105,6 +107,7 @@ private VespaVersion vespaVersionFromSlime(Inspector object) {
                                 Instant.ofEpochMilli(object.field(committedAtField).asLong()),
                                 object.field(isControllerVersionField).asBool(),
                                 object.field(isSystemVersionField).asBool(),
+                                object.field(isReleasedField).asBool(),
```
Comment (bratseth) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/VersionStatusSerializer.java]: You're ok with the default here if missing?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/versions/MavenRepositoryClient.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class MavenRepositoryClient implements MavenRepository {
```
Comment (bratseth) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/versions/MavenRepositoryClient.java]: Author and a comment please

**Review by bratseth — COMMENTED** [review]
Great!

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/VersionStatusSerializer.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/VersionStatusSerializer.java]: While the controllers upgrade to include this for the first time, the upgraded ones may fail to provide a compile version some of the time. I think this is acceptable.

**Pull request merged**